### DISCUSSION TITLE
How to prevent `CFLAGS` from being passed to linker?

### DISCUSSION THREAD

**Discussion by ywwry66** [libuv/libuv#3966] [Q&A] [answered]
Hi there, I ran into a problem when building R package `fs` which vendors `libuv`. I have `-Xpreprocessor -fopenmp` in `CPPFLAGS` for building R packages, which is then passed to `CFLAGS` for building `libuv`. This causes linking failure
```
clang: error: unsupported option '-fopenmp'
make[1]: *** [libuv.la] Error 1
make: *** [libuv-1.44.2/.libs/libuv.a] Error 2
```
To understand why the error occurs, I looked at the `Makefile` of `libuv` and see the following lines:
```make
libuv_la_LINK = $(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) \
	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(libuv_la_CFLAGS) \
	$(CFLAGS) $(libuv_la_LDFLAGS) $(LDFLAGS) -o $@
```
Is it intended that `libuv_la_CFLAGS` and `CFLAGS` are passed to the linker? If not, how to prevent them from being added?

Thanks for your help!

**Comment by bnoordhuis** [comment, marked as answer]
You're asking in the wrong place. If that fs package vendors libuv, you should get in touch with them, because you're essentially asking about the integration between the two.

Supporting evidence: libuv doesn't ship a Makefile, only a Makefile.am, and that file doesn't contain the stanza (or a template of the stanza) you mention.

**Reply by ywwry66** [reply]
Thanks for the reply. The reason I asked here is the release version https://dist.libuv.org/dist/v1.44.2/libuv-v1.44.2-dist.tar.gz has the stanza I mentioned in `Makefile.in`.

But you are right, the stanza is generated by `automake`, and there have been confusions aroused from this behavior in their mailing list:
https://lists.gnu.org/archive/html/automake/2016-06/msg00012.html
https://lists.gnu.org/archive/html/automake/2003-01/msg00036.html
The gist is that [GNU Coding Standards](https://www.gnu.org/prep/standards/html_node/Command-Variables.html) suggests passing `CFLAGS` to the linker.
```
CFLAGS should be used in every invocation of the C compiler, both those which do compilation and those which do linking.
```

So yes, I will open an issue with `fs`.
